pkg/scalers/pulsar_scaler.go

```diff
@@ -0,0 +1,301 @@
+	Msgthroughputout  float64                       `json:"msgThroughputOut"`
+	Bytesincounter    int                           `json:"bytesInCounter"`
+	Msgincounter      int                           `json:"msgInCounter"`
+	Bytesoutcounter   int                           `json:"bytesOutCounter"`
+	Msgoutcounter     int                           `json:"msgOutCounter"`
+	Averagemsgsize    float64                       `json:"averageMsgSize"`
+	Msgchunkpublished bool                          `json:"msgChunkPublished"`
+	Storagesize       int                           `json:"storageSize"`
+	Backlogsize       int                           `json:"backlogSize"`
+	Publishers        []interface{}                 `json:"publishers"`
+	Subscriptions     map[string]pulsarSubscription `json:"subscriptions"`
+	Replication       struct {
+	} `json:"replication"`
+	Deduplicationstatus string `json:"deduplicationStatus"`
+}
+
+// NewPulsarScaler creates a new PulsarScaler
+func NewPulsarScaler(config *ScalerConfig) (Scaler, error) {
+	pulsarMetadata, err := parsePulsarMetadata(config)
+	if err != nil {
+		pulsarLog.Error(err, "error parsing pulsar metadata")
+		return nil, fmt.Errorf("error parsing pulsar metadata: %s", err)
+	}
+
+	client := kedautil.CreateHTTPClient(config.GlobalHTTPTimeout, false)
+
+	if pulsarMetadata.enableTLS {
+		config, err := kedautil.NewTLSConfig(pulsarMetadata.cert, pulsarMetadata.key, pulsarMetadata.ca)
+		if err != nil {
+			return nil, err
+		}
+		client.Transport = &http.Transport{TLSClientConfig: config}
+	}
+
+	return &pulsarScaler{
+		client:   client,
+		metadata: pulsarMetadata,
+	}, nil
+}
+
+func parsePulsarMetadata(config *ScalerConfig) (pulsarMetadata, error) {
+	meta := pulsarMetadata{}
+	switch {
+	case config.TriggerMetadata["adminURLFromEnv"] != "":
+		meta.adminURL = config.ResolvedEnv[config.TriggerMetadata["adminURLFromEnv"]]
+	case config.TriggerMetadata["adminURL"] != "":
+		meta.adminURL = config.TriggerMetadata["adminURL"]
+	default:
+		return meta, errors.New("no adminURL given")
+	}
+
+	switch {
+	case config.TriggerMetadata["topicFromEnv"] != "":
+		meta.topic = config.ResolvedEnv[config.TriggerMetadata["topicFromEnv"]]
+	case config.TriggerMetadata["topic"] != "":
+		meta.topic = config.TriggerMetadata["topic"]
+	default:
+		return meta, errors.New("no topic given")
+	}
+
+	topic := strings.ReplaceAll(meta.topic, "persistent://", "")
+	meta.statsURL = meta.adminURL + "/admin/v2/persistent/" + topic + "/stats"
+
+	switch {
+	case config.TriggerMetadata["subscriptionFromEnv"] != "":
+		meta.subscription = config.ResolvedEnv[config.TriggerMetadata["subscriptionFromEnv"]]
+	case config.TriggerMetadata["subscription"] != "":
+		meta.subscription = config.TriggerMetadata["subscription"]
+	default:
+		return meta, errors.New("no subscription given")
+	}
+
+	meta.activationTargetQueryValue = 0
+	if val, ok := config.TriggerMetadata["activationTargetQueryValue"]; ok {
```

Comment:
if the target value is `msgBacklog`, based on the [convention defined](https://keda.sh/docs/2.8/concepts/scaling-deployments/#activating-and-scaling-thresholds), the key should be `activationMsgBacklog`